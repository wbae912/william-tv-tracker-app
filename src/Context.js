import React from 'react';

const tvContext = React.createContext({
  shows: [],
  deleteTvShow: () => {},
  addTvShow: () => {}
})

export default tvContext;