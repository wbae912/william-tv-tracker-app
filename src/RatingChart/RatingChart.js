import React, { Component } from 'react';
import { HorizontalBar } from 'react-chartjs-2';
import tvContext from '../Context';

export default class RatingChart extends Component {
  static contextType = tvContext;

  render() {
    const countFiveStars = this.context.shows.filter(show => show.rating === 5).length;
    const countFourStars = this.context.shows.filter(show => show.rating === 4).length;
    const countThreeStars = this.context.shows.filter(show => show.rating === 3).length;
    const countTwoStars = this.context.shows.filter(show => show.rating === 2).length;
    const countOneStar = this.context.shows.filter(show => show.rating === 1).length;
    const notRated = this.context.shows.filter(show => show.rating === 0).length;  
    const data = {
    labels: ['Not Rated', 'One Star', 'Two Stars', 'Three Stars', 'Four Stars', 'Five Stars'],
    datasets: [
      {
        label: 'TV Shows per Rating',
        backgroundColor: 'rgba(59, 45, 255, 0.2)',
        borderColor: 'rgba(59, 45, 255, 0.2)',
        borderWidth: 1,
        hoverBackgroundColor: 'rgba(255,99,132,0.4)',
        hoverBorderColor: 'rgba(255,99,132,1)',
        data: [notRated, countOneStar, countTwoStars, countThreeStars, countFourStars, countFiveStars]
      }
    ]
  };
    return (
      <div style={{width: '50%'}}>
        <HorizontalBar 
          data={data}
          height={300}
          options={{
            responsive: true,
            maintainAspectRatio: false,
            scales: {
              xAxes: [{
                  ticks: { display: true, stepSize: 1, beginAtZero: true, min: 0 },
                  gridLines: {
                      display: true,
                      drawBorder: true
                  }
              }],
              yAxes: [{
                  ticks: { display: true },
                  gridLines: {
                      display: false,
                      drawBorder: true
                  }
              }]
          }
          }}
        />
      </div>
    )
  }
}