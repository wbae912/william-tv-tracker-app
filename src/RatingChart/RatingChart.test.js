import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Context from '../Context';
import RatingChart from './RatingChart';

describe(`RatingChart component`, () => {
  const shows = [
      {
        "id": "1",
        "tv_title": "Example",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      },
      {
        "id": "2",
        "tv_title": "Example 2",
        "status": "Completed",
        "season_number": "1",
        "episode_number": "1",
        "rating": "3",
        "genre": "Action",
        "description": "Lorem ipsum",
        "review": "Lorem ipsum"
      }
    ];
  
  it('renders component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={{shows}}><RatingChart /></Context.Provider>
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders a .RatingChart by default', () => {
    const wrapper = shallow(<Context.Provider value={{shows}}><RatingChart /></Context.Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})