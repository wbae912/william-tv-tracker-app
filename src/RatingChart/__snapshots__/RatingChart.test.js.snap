// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingChart component renders a .RatingChart by default 1`] = `<RatingChart />`;
